### DISCUSSION TITLE
Unable to Compile Shaders

### DISCUSSION THREAD

**Discussion by justinstroud-campmeeker** [load1n9/caviar#11] [Q&A] [answered]
Howdy...came across your lib, and I'm wanting to port an Android game I've been working on to Deno/TS using a GL window (no browser.) Whenever I attempt to run this example (or *any* example):
```JavaScript
import { Rectangle, Scene, World } from "https://deno.land/x/caviar/mod.ts";
class Game extends Scene {
  setup() {
    const rect = new Rectangle(20, 20, 50, 50, "#0000ff");
    this.addChild(rect);
  }
}
const test = new World({
  title: "test",
  width: 800,
  height: 600,
  resizable: true,
}, [Game]);

await test.start();
```
I get the following error:
```
error: Uncaught (in promise) Error: An error occurred compiling the shaders: 
      throw new Error(`An error occurred compiling the shaders: ${msg}`);
            ^
    at loadShader (https://deno.land/x/caviar@2.6.9/src/renderers/webgl/util.ts:47:13)
    at initShaderProgram (https://deno.land/x/caviar@2.6.9/src/renderers/webgl/util.ts:16:26)
    at new WebGLRenderer2D (https://deno.land/x/caviar@2.6.9/src/renderers/webgl/2d.ts:40:21)
    at new World (https://deno.land/x/caviar@2.6.9/src/World.ts:75:21)
    at file:///C:/repos/vexts/main.ts:8:14
```

Any ideas? This isn't a JavaScript only/TypeScript only issue. It happens if I attempt to use any display mechanism other than the browser.

_EDIT: I'm pretty unfamiliar with the whole non-browser-GL thing, but I am more than comfortable learning something new._

**Comment by load1n9** [comment, marked as answer]
caviar is a little rusty right now ill have to move it from GL to WebGPU in deno, I would check out https://github.com/deno-windowing/dwm for the canvas API for now

**Reply by justinstroud-campmeeker** [reply]
Thanks! I already started checking it out.
